jobs/ci-run/integration/common/test-runner.sh

```diff
@@ -32,6 +32,23 @@ while sudo lsof /var/lib/apt/lists/lock 2> /dev/null; do
 done
 sudo apt-get -y update
 
+setup_parca_agent() {
+    # Parca agent is already installed, then skip it.
+    if [ "$(which parca-agent >/dev/null 2>&1)" ]; then
+        return
+    fi
+
+    sudo snap install parca-agent --classic
+    sudo snap set parca-agent metadata-external-labels="machine=ci-run-${{BOOTSTRAP_PROVIDER}}"
```

Comment:
Just a thought, we could add some labels like `TEST_RUNNER_NAME` to make it easier to correlate to issues we eventually find?

Reply:
Maybe, I think we just care about the provider/substrate rather what test is running. Otherwise we might start gaming for a specific test and not juju as a whole.

Reply:
AIUI, this actually sets it up on the host whenever we have an lxd run, is that correct? So this isn't setting up parca-agent *in* lxd, it is setting up parca-agent on hosts that run lxd.
I have a slight concern about side effects if you are trying to run the CI suite locally. In general, we might want to have a CI step for Jenkins ephemeral bots to install and configure parca-agent, but I'm kind of against doing it as a side effect of running a test.